src/app/tools/active_tool.cpp

```diff
@@ -237,6 +237,15 @@ void ActiveToolManager::setSelectedTool(Tool* tool)
 bool ActiveToolManager::isToolAffectedByRightClickMode(Tool* tool)
 {
   bool shadingMode = (Preferences::instance().tool(tool).ink() == InkType::SHADING);
+
+  if (shadingMode) {
+
+    ColorBar* colorbar = ColorBar::instance();
+    if (colorbar->getPaletteView()->getLastSelectionSize() < 2)
+      shadingMode = false;
```

Comment:
Be aware that the user can change the selected shades from the `ContextBar` itself:

![image](https://github.com/user-attachments/assets/01f25c08-7615-423e-b51c-e9823587fce7)

The information of the active shades can be accessed through the `ContextBar::getShade()` member function.